pkg/operator/operator.go

```diff
@@ -544,10 +544,17 @@ func (o *Operator) sync(key string) error {
 		klog.Errorf("error occurred while setting status to in progress: %v", err)
 	}
 
-	taskName, err := tl.RunAll()
-	if err != nil {
-		o.reportError(err, taskName)
-		return err
+	taskErrors := tl.RunAll()
+	if len(taskErrors) > 0 {
+		var failedTask string
+		if len(taskErrors) == 1 {
+			failedTask = strings.Join(strings.Fields(taskErrors[0].Name+"Failed"), "")
```

Comment:
so this isn't related to your change but I'm pretty sure that this line isn't correct... It should rather be

```suggestion
			failedTask = cmostr.ToPascalCase(taskErrors[0].Name+"Failed")
```

Reply:
This is happening in the reportError method.